File: Languages/C#/BubbleSort.cs
// Source: ChatGPT

using System;

class BubbleSort
{
    public static bool LOGGING = false;
    
    static void Main(string[] args)
    {
        int[] arr = {44411,84470,23178,48948,15058,16444,37478,32140,605,67448};
        int n = arr.Length;
        BubbleSortMethod(arr, n);
        if (LOGGING)
        {
            Console.WriteLine("Sorted array:");
            PrintArray(arr);
        }

    }

    static void BubbleSortMethod(int[] arr, int n)
    {
        for (int i = 0; i < n - 1; i++)
        {
            for (int j = 0; j < n - i - 1; j++)
            {
                if (arr[j] > arr[j + 1])
                {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    static void PrintArray(int[] arr)
    {
        int n = arr.Length;
        for (int i = 0; i < n; ++i)
            Console.Write(arr[i] + " ");
        Console.WriteLine();
    }
}

File: Languages/Haskell/BubbleSort.hs
-- Source: ChatGPT
logging = False

bubbleSort :: Ord a => [a] -> [a]
bubbleSort [] = []
bubbleSort xs = bubbleSort' xs (length xs)

bubbleSort' :: Ord a => [a] -> Int -> [a]
bubbleSort' xs 0 = xs
bubbleSort' xs n = bubbleSort' (bubble xs) (n-1)

bubble :: Ord a => [a] -> [a]
bubble [] = []
bubble [x] = [x]
bubble (x:y:xs) 
    | x > y     = y : bubble (x:xs)
    | otherwise = x : bubble (y:xs)


main :: IO ()
main = do
  let unsortedList = [10650,56341,77676,47158,34999,22009,19680,41870,31090,30668]
      sortedList = bubbleSort unsortedList
  if (logging)
    then putStrLn ("Bubble Sorted list: " ++ show sortedList)
    else print (length(sortedList))
